system/jlib/jptree.cpp

```diff
@@ -3132,6 +3132,7 @@ void PTree::addLocal(size32_t l, const void *data, bool _binary, int pos)
         IPropertyTree *element1 = detach();
         array = new CPTArray();
         addingNewElement(*element1, ANE_APPEND);
+        static_cast<PTree *>(element1)->setOwner(array);
```

Comment:
looks like a good fix, curious what code cause it to be exposed?

Reply:
Two things.  When you exported to YAML it generated the first item as a nested structure, and the second as a list.  E.g.

hosts:
  ip: x
hosts:
- ip: x

Also the mergeConfig code misinterpreted the structure.

Reply:
ok thanks.